plugins/chromatic_correction.py

```diff
@@ -0,0 +1,222 @@
+    """
+    # Calculate pairwise distances
+    distances = cdist(centers1, centers2)
+
+    # Find pairs within the max_distance
+    pairs = numpy.argwhere(distances <= max_distance)
+
+    return pairs
+
+
+class ChromaticCorrectionPlugin(Plugin):
+    name = "Chromatic Correction"
+    __version__ = "0.1"
+    __author__ = u"Karishma Kumar"
+    __license__ = "GPLv2"
+
+    def __init__(self, microscope, main_app):
+        super(ChromaticCorrectionPlugin, self).__init__(microscope, main_app)
+        self.addMenu("Data correction/Lateral chromatic correction", self.start)
+
+    def show_static_stream(self, filename: str, data: model.DataArray):
+        """
+        Save the given data, read and display in the analysis tab
+        :param filename: filename of the data array
+        :param data: value and metadata of the given data array
+        """
+        # display the reference in the analysis tab
+        analysis_tab = self.main_app.main_data.getTabByName('analysis')
+        # save the image
+        dataio.tiff.export(filename, data)
+        # Read the saved images and display
+        im_read = dataio.tiff.read_data(filename)
+        aligned_stream = StaticFluoStream(im_read[0].metadata[model.MD_DESCRIPTION], im_read[0])
+        scont = analysis_tab.stream_bar_controller.addStream(aligned_stream, add_to_view=True)
+        scont.stream_panel.show_remove_btn(True)
+
+    def start(self):
+        tab_data = self.main_app.main_data.tab.value.tab_data_model
+        self.new = []
+        self.ref = []
+        levels = generate_zlevels(tab_data.main.focus,
+                                  (-10e-06, 10e-06),
+                                  5e-06)
+
+        non_static = [s for s in tab_data.streams.value
+                      if isinstance(s, FluoStream)]
+        zlevels = {s: levels for s in non_static}
+        settings_observer = SettingsObserver(model.getComponents())
+        acqui_task = acquireZStack(non_static, zlevels, settings_obs=settings_observer)
+        das, e = acqui_task.result()
+
+        if len(das) <= 1:
+            box = wx.MessageDialog(self.main_app.main_frame,
+                                   "Add minimum two streams. One of the streams must be green channel.",
+                                   "Failed to do chromatic correction", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # Calculate Maximum Intensity projection for all the present channels in the stream panel
+        for i, da in enumerate(das):
+            mip_image = model.DataArray(numpy.amax(da, axis=0), copy.copy(da.metadata))
+            dataio.tiff.export(f"{da.metadata[model.MD_DESCRIPTION]}_{i}", mip_image)
+            # All the channels will be corrected according to a reference colored channel
+            # green channel is selected as reference channel, Why? -> Ask Deniz
+            if da.metadata[model.MD_USER_TINT] == (0, 255, 0):
```

Comment:
I think this should be based off the emission/excition rather than the colour tint. It might be best to let the user select the channel somehow?

Reply:
I'd say it's not so bad, because using tint allows the user to select any channel (as the tint can be user selectable). As this plugin is to be used only internally, once per system, I think it's a good trade-off between development time and flexibility. 
Would be worth mentioning this in the comment.